File: food-donation-frontend/src/services/authService.js
import apiClient from './apiClient';

const authService = {
  async register(payload) {
    const { data } = await apiClient.post('/auth/register', payload);
    return data.data;
  },
  async login(credentials) {
    const { data } = await apiClient.post('/auth/login', credentials);
    return data.data;
  },
  async logout() {
    const { data } = await apiClient.post('/auth/logout');
    return data;
  },
  async forgotPassword(email) {
    const { data } = await apiClient.post('/auth/forgot-password', { email });
    return data;
  },
  async resetPassword(payload) {
    const { data } = await apiClient.post('/auth/reset-password', payload);
    return data;
  },
  async verifyEmail(token) {
    const { data } = await apiClient.get(`/auth/verify-email?token=${token}`);
    return data;
  },
};

export default authService;

File: food-donation-backend/src/main/java/com/fooddonation/service/AuthService.java
package com.fooddonation.service;

import com.fooddonation.dto.request.ForgotPasswordRequest;
import com.fooddonation.dto.request.LoginRequest;
import com.fooddonation.dto.request.RegisterRequest;
import com.fooddonation.dto.request.ResetPasswordRequest;
import com.fooddonation.dto.response.AuthResponse;

public interface AuthService {

	AuthResponse register(RegisterRequest request);

	AuthResponse login(LoginRequest request);

	AuthResponse refreshToken(String refreshToken);

	void logout(String token);

	void forgotPassword(ForgotPasswordRequest request);

	void resetPassword(ResetPasswordRequest request);

	void verifyEmail(String token);

}
